Gems/AWSClientAuth/cdk/utils/resource_name_sanitizer.py

```diff
@@ -0,0 +1,45 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+import hashlib
+from aws_cdk import (
+    core,
+    aws_cognito as cognito,
+    aws_iam as iam
+)
+
+MAX_RESOURCE_NAME_LENGTH_MAPPING = {
+    core.Stack.__name__: 128,
+    iam.Role.__name__: 64,
+    iam.ManagedPolicy.__name__: 144,
+    cognito.CfnUserPoolClient.__name__: 128,
+    cognito.CfnUserPool.__name__: 128,
+    cognito.CfnIdentityPool.__name__: 128
+
+}
+
+
+def sanitize_resource_name(resource_name: str, resource_type: str) -> str:

```

Comment:
Minor: Slight concern that this violates DRY. But right now we do not have a common area for CDK utils.

Reply:
Yeah, we have a Jira task for linking CDK applications for different gems but it hasn't been done yet.